crates/primitives/src/alloy_compat.rs

```diff
@@ -123,7 +123,14 @@ impl TryFrom<alloy_rpc_types::Transaction> for Transaction {
                     ))
                 }
                 Ok(Self::Legacy(TxLegacy {
-                    chain_id: tx.chain_id,
+                    chain_id: tx.chain_id.or_else(|| {
+                        tx.signature.and_then(|signature| {
+                            // TODO: make this better, this is needed because sometimes rpc returns
+                            // legacy transactions without a chain id explicitly in the response,
+                            // however it would have a chain id in the signature from eip155
+                            extract_chain_id(signature.v.to()).expect("just dont error").1
+                        })
+                    }),
```

Comment:
Is this a bug we should fix outside of this PR? cc @mattsse

Reply:
probably, this is more of a quirk w.r.t. RPC endpoints that don't act like reth or geth, which will usually populate the chain id